syncd/scripts/persistent_mloop.py

```diff
@@ -0,0 +1,171 @@
+import time
+import shutil
+import os.path
+from sonic_py_common.general import check_output_pipe
+
+CONFIGURED_PORTS_PATH = "/etc/mloop_conf/"
+CONFIGURED_PORTS_FILE = "mloop_ports.json"
+SAISDKDUMP_PATH = "/saisdkdump_file"
+SERVICE_FILE = "persistent_mloop.conf"
+SERVICE_PATH = "/etc/supervisor/conf.d/"
+
+def port_sorting(port):
+    """
+    For translation dict sorting. Returns the port number in port name.
+    """
+    return int(port[0][len("Ethernet"):]) if "Ethernet" in port[0] else 0
+
+class MloopConfig:
+    def __init__(self):
+        self.ports = []
+        self.build_translation_dict()
+
+    def config_ports(self, ports=None):
+        configured_ports = []
+
+        if ports:
+            self.ports = ports
+
+        # The first time doesn't work for some reason
+        first_port = self.port_translation.get(self.ports[0])
+
+        if first_port:
+            self.config_port_to_mloop(first_port)
+
+        for port in self.ports:
+            logical_port = self.port_translation.get(port)
+
+            if not logical_port:
+                print(f"{port} doesn't exist")
+                continue
+
+            self.config_port_to_mloop(logical_port)
+            configured_ports.append(port)
+
+    def config_range(self, port_range):
+        logical_ports = self.parse_range(port_range)
+
+        if not logical_ports:
+            print("Invalid port range")
+            return
+        
+        for logical_port in logical_ports:
+            self.config_port_to_mloop(logical_port)
+
+        self.save_mloop_ports()
+    
+    def build_translation_dict(self):
+        self.port_translation = {}
+
+        subprocess.run(["saisdkdump", "-f", SAISDKDUMP_PATH], shell=False, stdout=subprocess.PIPE, text=True)
+
+        with open(SAISDKDUMP_PATH, 'r') as saisdk_file:
+            dump_lines = saisdk_file.read()
+
+        start_index = dump_lines.find("netdev_dump")
+        end_index = dump_lines.find("cmd_ifc_dump")
+
+        port_table = dump_lines[start_index:end_index]
+        port_table = port_table.split('\n')
+        port_table = port_table[4:]
+
+        for line in port_table:
+            line = line.split()
+            self.port_translation[line[1]] = line[2]
+
+        self.port_translation = dict(sorted(self.port_translation.items(), key=port_sorting))
+        
+    def save_mloop_ports(self):
+        os.makedirs(CONFIGURED_PORTS_PATH, exist_ok=True)
+        full_path = os.path.join(CONFIGURED_PORTS_PATH, CONFIGURED_PORTS_FILE)
+        with open(full_path, 'w') as ports_file:
+            json.dump(self.ports, ports_file)
+
+    def config_port_to_mloop(self, logical_port):
+        configured = False
+        retries = 0    
+        subprocess.run(["sx_api_port_phys_loopback.py", "--cmd", "0", "--log_port", logical_port, "--loopback_type", "2", "--force" ], 
+                        shell=False, stdout=subprocess.PIPE, text=True)
+        while (not configured) and (retries < 500):
```

Comment:
500 retries is too much. If this is only to make sure you are starting after SDK is ready - we need to delay the service or make sure it is ready before we start.